File: drawee-backends/drawee-pipeline/src/main/java/com/facebook/drawee/backends/pipeline/info/ImagePerfDataListener.java
/*
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
package com.facebook.drawee.backends.pipeline.info;

public interface ImagePerfDataListener {

  void onImageLoadStatusUpdated(ImagePerfData imagePerfData, @ImageLoadStatus int imageLoadStatus);

  void onImageVisibilityUpdated(ImagePerfData imagePerfData, @VisibilityState int visibilityState);
}

File: ui-common/src/main/java/com/facebook/fresco/ui/common/ImagePerfDataListener.kt
/*
 * Copyright (c) Meta Platforms, Inc. and affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

package com.facebook.fresco.ui.common

interface ImagePerfDataListener {

  fun onImageLoadStatusUpdated(imagePerfData: ImagePerfData, imageLoadStatus: ImageLoadStatus)

  fun onImageVisibilityUpdated(imagePerfData: ImagePerfData, visibilityState: VisibilityState)
}
